app/templates/components/public/ticket-list.hbs

```diff
@@ -198,7 +198,6 @@
             {{#if this.device.isMobile}}
               <div class="ui hidden divider"></div>
             {{/if}}
-            <a href="#" {{action 'togglePromotionalCode'}}>{{t 'Enter promotional code'}}</a>
```

Comment:
no need to remove it from the component. fix the function `hasOnlyFreeTickets`.

Reply:
What is its use there? It just points to _#_

Reply:
It should not be shown when there are only free tickets.

Reply:
I am saying that keep action and toggle promotion in same component according to best coding practices. Otherwise writing working code isn't a big thing